src/main/java/Parser.java

```diff
@@ -0,0 +1,114 @@
+import java.time.LocalDate;
+import java.util.Scanner;
+
+/**
+ * Listener of inputs
+ */
+public class Parser {
+
+
+    public String listenerForUI(TaskList myTasklist, Ui myUi,String inputUI) {
+
+        String currInput = inputUI;
+
+        String response= "";
+            myUi.horizontalRule();
+            String[] fullArg = currInput.split("/");
+            String[] args1 = fullArg[0].split(" ");
+
+
+            if (currInput.equals("bye")) {
+                response += "Bye. Hope to see you again soon!\n";
+                myUi.horizontalRule();
+            }
+
+            if (args1[0].equals("find")) {
+                String findItem = "";
+                for (int i = 1; i < args1.length; i++) {
+                    findItem += args1[i] + " ";
+                }
+                response+=myTasklist.findTaskUI(findItem);
+            } else if (args1[0].equals("todo")) {
+                String task = "";
+                for (int i = 1; i < args1.length; i++) {
+                    task += args1[i] + " ";
+                }
+                try {
+                    response+=myTasklist.addTaskUI("[T]", task);
+                } catch (InSuffArgsException E) {
+                    System.out.println("☹ OOPS!!! The description of a " + args1[0] + " cannot be empty.");
```

Comment:
In relation to my comment about this exception class. Maybe overriding the toString method of this class to print the error message would be better? I understand you need the args1[0] variable also, maybe you can pass it along as an argument in the constructor of the exception?